README.md

```diff
@@ -38,6 +38,72 @@ If you want to test what happens when the HTTP server goes down, use `Bypass.dow
 TCP socket and `Bypass.up/1` to start listening on the same port again. Both functions block until
 the socket updates its state.
 
+### Expect Functions
+
+You can take any of the following approaches:
+* `expect/2` or `expect_once/2` to install a generic function that all calls to bypass will use
+* `expect/4` and/or `expect_once/4` to install specific routes (method, path combinations)
+* a combination of the above, where the routes will be used first, and then the generic version
+  will be used as default
+
+#### expect/2 (bypass_instance, function)
+
+Must be called at least once, and it will error if not.  As an alternative, `nil` can be passed
```

Comment:
I would like to reduce the number of possible outcomes one can achieve from all those functions. Supporting `nil` introduces some overlapping functionality: `expect(bypass, nil)` and `expect_once(bypass, nil)` have the same meaning.

It's better not to support `nil` in those functions. The user can provide a function that always fails and include any desired information in it. For example,
```
Bypass.expect(bypass, fn conn ->
  flunk "Got an unexpected request at #{conn.req_path}"
end)
```

is more informative than a generic `Bypass instance <instance> got an unexpected request` error.